Pachanga/Core/Firestore/MatchManager.swift

```diff
@@ -0,0 +1,145 @@
+//
+//  MatchManager.swift
+//  Pachanga
+//
+//  Created by Javier Alaves on 21/8/23.
+//
+
+import FirebaseFirestore
+import FirebaseFirestoreSwift
+import Foundation
+import SwiftUI
+
+struct Match: Codable {
+    
+    // props
+    let matchId: String
+    let dateCreated: Date
+    var location: String
+    var matchDate: Date
+    var players: [String] // combo of teamOne and teamTwo
+    var teamOne: [String] // tuple instead?
+    var teamTwo: [String]
+    var teamOneScore: Int
+    var teamTwoScore: Int
+    var isRanked: Bool // whether match should contribute to rankings or not
+    
+    // base init
+    init(
+        location: String,
+        matchDate: Date,
+        players: [String],
+        teamOne: [String],
+        teamTwo: [String],
+        teamOneScore: Int,
+        teamTwoScore: Int,
+        isRanked: Bool
+    ) {
+        self.matchId = UUID().uuidString
+        self.dateCreated = Date.now
```

Comment:
Link to session date, honestly might not be worth registering at all since it's always going to be equivalent to it